### PR TITLE
#10 - Add Multiple Ldap Hosts

### PR DESCRIPTION
**Pull request by jmsolutionsgroup**
Create a way to add multiple ldap hosts and allow the end user which domain to authenticate against

### CODE REVIEW

**Review by jmsolutionsgroup — COMMENTED** [review]
This adds the ability to add multiple hosts by changing the structure of the JSON in the appsettings.  

Currently it only allows multiple hosts of the same type of user, ie ActiveDirectoryAppUser or OpenLdapAppUser

**Comment by Nordes** [comment]
I guess my e-mail was not pushed over here ;). Sorry for the late reply. As I was mentioning in my mail, this have been requested by somebody else, but I had no time to investigate. I was proposing to use regular expression in order to discriminate what AD query needs to be done. 

I will now look at the pull request ;). It might takes a few day due to my busy daily workload.

> **Note:** I just saw that you added a dropdown in order to choose the domain. However, this will cause an issue when you do the API calls to IdentityServer in order to login from a front-end application (get the authentication token). I'll look more in depth in the pull request later probably this weekend)

Btw: great work.

File: IdentityServer.LdapExtension.Unit/LdapServiceTests.cs [hunk 1]
```diff
@@ -14,13 +15,20 @@ public class LdapServiceTests
...
+                new LdapHost {
+                    Name= "default",
+                    Url = "localhost",
+                    BindDn = "cn=ldap-ro,dc=contoso,dc=com",
+                    BindCredentials = "P@ss1W0Rd!",
+                    SearchBase = "ou=users,DC=contoso,dc=com",
+                    SearchFilter = "(&(objectClass=posixAccount)(objectClass=person)(uid={0}))",
+                    Ssl = false,
+                    Port = 389
+
```
Comment (Nordes) [inline IdentityServer.LdapExtension.Unit/LdapServiceTests.cs]: no need of the empty line

File: IdentityServer.LdapExtension/ILdapService.cs [hunk 2]
```diff
@@ -15,12 +15,27 @@ public interface ILdapService<out TUser>
         /// <param name="password">The password.</param>
         /// <returns>Returns the logged in user.</returns>
         TUser Login(string username, string password);
+        /// <summary>
```
Comment (Nordes) [inline IdentityServer.LdapExtension/ILdapService.cs:19]: keep the style for the spacing ;). One blank line before the summary of a method.

**Comment by murattdogan** [comment]
Not Working Object reference not set to an instance of an object @uchetfield uchetfield
Ldap Timeout
adsiz
var user = searchResult.next(); old proje working 
@uchetfield  your project  var user = searchResult.next(); new proje not  working  pleease 
I need help  muratdogan@hitit.edu.tr

**Comment by murattdogan** [comment]
.AddLdapUsers<ActiveDirectoryAppUser>(Configuration.GetSection("multipleActiveDirectory"), UserStore.InMemory);

Multiple multipleActiveDirectory or OpenLdapAppUser   MultipleLdapAppUser ?

**Comment by Nordes** [comment]
I was re-looking into your pull request. There's some things that I like. Please look at what I've done in the branch `features/multi_ldap`. (I will delete the branch once it's merged in develop. 

Basically, I keep the retro-compatibility with the previous version. This way, there's no breaking changes.

**Comment by jmsolutionsgroup** [comment]
I'm going to close this pull request because @Nordes  created a better option

**Pull request closed without merging**